main/zoe/guide/working_with_ZCFSeats.md

```diff
@@ -0,0 +1,229 @@
+# Seat objects
+Zoe uses *seats* to access or manipulate offers. They let
+contracts and users interact  with *active offers*. A seat has two
+facets, which we call a `UserSeat` and a `ZCFSeat`, which are
+two ways of accessing an offer's information.
+
+- `ZCFSeats` are used within contracts and with ZCF methods. They
+handle reallocation of assets between users and are a representation of an
+offer by the contract. 
+
+- `UserSeats` represent offers external to Zoe and the contract.. They are
+the representation of the offer on the user side and let a contract instance
+see what the user has escrowed and ensure when the user gets a payout, 
+they get the right amount. 
+
+For example, in an auction, the person putting up the item has both a `ZCFSeat`,
+and a `UserSeat` as do each of the bidders. When the auction ends, the contract instance uses these `ZCFSeats` to reallocate the assets between the seller and the  winning bidder. The bidders use their `UserSeats` to check in on what's happening 
+with the auction contract instance. When the auction ends, `UserSeat` allocations
+are changed to reflect the resulting asset allocation, and payouts go to the `UserSeats`.
+
+## ZCFSeats
+
+A `ZCFSeat` can synchronously query its
+associated offer's current state, such as its currently
+allocated asset amounts. It can also synchronously 
+manipulate the offer.
+
+`ZCFSeats` have a *staged allocation*.  This is the current allocation
+value used if and when the seat is *reallocated*  over. Reallocation occurs
+when an offer successfully exits and the assets involved are
+redistributed to the contract instance participants based on the
+instance's result. 
+
+A seat's staged allocation starts out as empty. An empty allocation
+means when the offer resolves, no assets are distributed via this
+seat. **tyg todo: Is "empty" the right word? The operation is "clear",
+but I think "empty" conveys better the effect at least** There are
+operations to add and subtract amounts from a seat's staged
+allocation as offers are made and rejected, as well as to clear it
+back to its initial empty state.  
+
+A `ZCFSeat` has methods that do the following:
+- Create and exit a seat.
+- Get the seat's 
+   - Offer proposal. 
+   - Amount allocated for one keyword. 
+   - Current allocation for all keywords. 
+   - Current staged allocation.
+- Check if a proposed allocation satisfies offer safety. 
+- Manipulate a seat's staged allocation:
+   - Check if the seat has a non-empty staged allocation.
+   - Add an amount to the staged allocation.
+   - Subtract an amount from the staged allocation.
+   - Clear the staged allocation.
+- Associate a notifier with the seat's allocation.
+
+Also, `zcf.reallocate()` is called over an offer's `ZCFseats` with
+non-empty staged allocations. 
+
+### Creating and Exiting a `ZCFSeat`
+
+#### Creating a `ZCFSeat`
+
+To create a new, empty, `ZCFSeat`, use `zcf.makeEmptySeatKit()`. It
```

Comment:
The most common case (probably over 90% of cases) is that the ZCFSeat is sent to the contract by ZCF as the argument to the offerHandler for that offer. I think we should not lead with `makeEmptySeatKit` here. That is only for the case in which the contract wants to make another seat (without any offer safety or payments escrowed) for accounting purposes, such as the seat associated with a liquidity pool.